commands/operator-sdk/cmd/build.go

```diff
@@ -135,29 +136,31 @@ func renderTestManifest(image string) {
 const (
 	build      = "./tmp/build/build.sh"
 	configYaml = "./config/config.yaml"
+	mainGo     = "./cmd/%s/main.go"
 )
 
 func buildFunc(cmd *cobra.Command, args []string) {
 	if len(args) != 1 {
 		cmdError.ExitWithError(cmdError.ExitBadArgs, fmt.Errorf("build command needs exactly 1 argument"))
 	}
 
-	bcmd := exec.Command(build)
-	bcmd.Env = append(os.Environ(), fmt.Sprintf("TEST_LOCATION=%v", testLocationBuild))
-	bcmd.Env = append(bcmd.Env, fmt.Sprintf("ENABLE_TESTS=%v", enableTests))
-	o, err := bcmd.CombinedOutput()
-	if err != nil {
-		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to build: (%v)", string(o)))
+	// Don't need to buld go code if Ansible Operator
+	if buildCmd() {
+		bcmd := exec.Command(build)
+		o, err := bcmd.CombinedOutput()
```

Comment:
As @AlexNPavel mentioned: You need to pass the envs from L146-147 to the build command.
```Go
bcmd.Env = append(os.Environ(), fmt.Sprintf("TEST_LOCATION=%v", testLocationBuild))
bcmd.Env = append(bcmd.Env, fmt.Sprintf("ENABLE_TESTS=%v", enableTests))
```
We made this change in #469 

Although we're planning to remove the build script altogether in #541 so you might need to rebase again if that gets done first.